packages/charts/react-charting/src/components/DeclarativeChart/PlotlySchemaAdapter.ts

```diff
@@ -172,6 +177,96 @@ export const transformPlotlyJsonToGVBCProps = (
   };
 };
 
+// TODO: Add support for padding in continuous axis
+// FIXME: Vertical bar chart displays duplicate legend buttons
+export const transformPlotlyJsonToVBCProps = (
+  jsonObj: any,
+  colorMap: React.MutableRefObject<Map<string, string>>,
+): IVerticalBarChartProps => {
+  const { data, layout } = jsonObj;
+  const vbcData: IVerticalBarChartDataPoint[] = [];
+
+  data.forEach((series: any, index: number) => {
+    if (!series.x) {
+      return;
+    }
+
+    const scale = d3ScaleLinear()
+      .domain(d3Extent<number>(series.x) as [number, number])
+      .nice();
+    let [xMin, xMax] = scale.domain();
+
+    xMin = typeof series.xbins?.start === 'number' ? series.xbins.start : xMin;
+    xMax = typeof series.xbins?.end === 'number' ? series.xbins.end : xMax;
+
+    const bin = d3Bin().domain([xMin, xMax]);
+
+    if (typeof series.xbins?.size === 'number') {
+      const thresholds: number[] = [];
+      let th = xMin;
+      const precision = d3PrecisionFixed(series.xbins.size);
+      const format = d3Format(`.${precision}f`);
+
+      while (th < xMax + series.xbins.size) {
+        thresholds.push(parseFloat(format(th)));
+        th += series.xbins.size;
+      }
+
+      xMin = thresholds[0];
+      xMax = thresholds[thresholds.length - 1];
+      bin.domain([xMin, xMax]).thresholds(thresholds);
+    }
+
+    const buckets = bin(series.x);
+    const totalDataPoints = d3Merge(buckets).length;
+
+    buckets.forEach(bucket => {
+      if (bucket.length < 1) {
+        return;
+      }
+
+      const legend = series.name || `Series ${index + 1}`;
+      const color = getColor(legend, colorMap);
+      let y = bucket.length;
+
+      if (series.histnorm === 'percent') {
+        y = (bucket.length / totalDataPoints) * 100;
+      } else if (series.histnorm === 'probability') {
+        y = bucket.length / totalDataPoints;
```

Comment:
do the value of the bucket not matter for histogram probability

Reply:
No. Got these formulas from [here](https://plotly.com/javascript/reference/histogram/#histogram-histnorm) and also verified them with some examples.

Reply:
Here are their implementations. See if it matches with our asssumption.
https://github.com/plotly/plotly.js/blob/3cafc650c2ded5436bd0e7a5e4c63af3acaeeded/src/traces/histogram/norm_functions.js
https://github.com/plotly/plotly.js/blob/3cafc650c2ded5436bd0e7a5e4c63af3acaeeded/src/traces/histogram/bin_functions.js
https://github.com/plotly/plotly.js/blob/3cafc650c2ded5436bd0e7a5e4c63af3acaeeded/src/traces/histogram/calc.js

Reply:
Yes, they look same.